src/peep/staff.c

```diff
@@ -367,6 +367,11 @@ void game_command_fire_staff_member(int *eax, int *ebx, int *ecx, int *edx, int
 		window_close_by_class(WC_FIRE_PROMPT);
 		uint16 sprite_id = *edx;
 		rct_peep *peep = &g_sprite_list[sprite_id].peep;
+		if (peep->sprite_identifier != SPRITE_IDENTIFIER_PEEP || peep->type != PEEP_TYPE_STAFF)
```

Comment:
Are you sure?
First check is if the sprite is peep or not, then there's a check if the peep is staff or not and we want both of these to be valid, as this is `game_command_fire_staff_member`.

Using De Morgan's first, this check equals:
`!(peep->sprite_identifier == SPRITE_IDENTIFIER_PEEP && peep->type == PEEP_TYPE_STAFF)`
which looks correct to me.